includes/Core/Util/Health_Checks.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Health_Checks
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Exception;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit_Dependencies\Google_Service_Webmasters;
+use WP_REST_Server;
+
+/**
+ * Class for performing health checks.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Health_Checks {
+
+	/**
+	 * Authentication instance.
+	 *
+	 * @var Authentication
+	 */
+	protected $authentication;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Authentication $authentication Authentication instance.
+	 */
+	public function __construct( Authentication $authentication ) {
+		$this->authentication = $authentication;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_filter(
+			'googlesitekit_rest_routes',
+			function ( $rest_routes ) {
+				$health_check_routes = $this->get_rest_routes();
+
+				return array_merge( $rest_routes, $health_check_routes );
+			}
+		);
+	}
+
+	/**
+	 * Gets all health check REST routes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return REST_Route[]
+	 */
+	private function get_rest_routes() {
+		return array(
+			new REST_Route(
+				'core/site/data/health-checks',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function() {
+							$checks = array(
+								'googleAPI' => $this->check_google_api(),
+							);
+
+							return compact( 'checks' );
+						},
+						'permission_callback' => function () {
+							return current_user_can( Permissions::SETUP );
+						},
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Checks connection to Google APIs.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Results data.
+	 */
+	private function check_google_api() {
+		$client        = $this->authentication->get_oauth_client()->get_client();
+		$restore_defer = $client->withDefer( false );
+		$error_msg     = '';
```

Comment:
A minor thing, but 
```suggestion
		$error_message = '';
```
is a clearer variable name.

Reply:
Thought, I suppose the array shape includes `errorMsg` already, so this is probably fine to leave as-is. 😆